hal/src/nRF52840/spi_hal.cpp

```diff
@@ -275,13 +275,18 @@ void HAL_SPI_Init(HAL_SPI_Interface spi) {
 }
 
 void HAL_SPI_Begin(HAL_SPI_Interface spi, uint16_t pin) {
+    HAL_SPI_Acquire(spi, nullptr);
```

Comment:
We did a similar thing (guarding each function with acquire/release) for I2C hal previously in order not to break the applications built with previous DeviceOS version, because we've started using PMIC and FuelGauge in power manager.

In case of SPI and Ethernet we've always had this issue, we are not introducing a potential problem, so let's just have proper locking in wiring and require the applications to be rebuilt.